walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,81 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  /* Finds all time ranges that satisfy the request given the already existing set of events. */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Find potential conflicts between scheduled events and request using only
+    // mandatory attendees.
+    Collection<String> mandatoryAttendees = request.getAttendees();
+    List<TimeRange> mandatoryAttendeeConflicts = getPotentialConflicts(events, mandatoryAttendees);
+
+    // Find potential conflicts between scheduled events and request using all attendees.
+    List<TimeRange> allConflicts = new ArrayList(mandatoryAttendeeConflicts);
+    allConflicts.addAll(getPotentialConflicts(events, request.getOptionalAttendees()));
+
+    // Find time ranges based on all conflicts.
+    Collection<TimeRange> timeRanges = queryUsingPotentialConflicts(events, request, allConflicts);
+
+    // If the number of time ranges is positive or the number of mandatory attendees is zero
```

Comment:
Make this comment about the more general case and not just what the code says? 
Eg // If optional scheduling conflicts with mandatory scheduling or If no mandatory attendees

Reply:
Done - yes I agree the comment restates the code.